lib/ansible/modules/cloud/vmware/vmware_guest_screenshot.py

```diff
@@ -0,0 +1,261 @@
+                handle.write(bytes_read)
+                handle.flush()
+                os.fsync(handle.fileno())
+                download_size += len(bytes_read)
+                bytes_read = response.read(2 ** 20)
+
+        return download_size
+
+    def get_screenshot_facts(self, task_info, file_url, file_size):
+        screenshot_facts = dict()
+        if task_info is not None:
+            screenshot_facts = dict(
+                virtual_machine=task_info.entityName,
+                screenshot_file=task_info.result,
+                task_start_time=task_info.startTime,
+                task_complete_time=task_info.completeTime,
+                result=task_info.state,
+                screenshot_file_url=file_url,
+                download_local_path=self.params.get('local_path'),
+                download_file_size=file_size,
+            )
+
+        return screenshot_facts
+
+    def take_vm_screenshot(self):
+        if self.current_vm_obj.runtime.powerState != vim.VirtualMachinePowerState.poweredOn:
+            self.module.fail_json(msg="VM is %s, valid power state is poweredOn." % self.current_vm_obj.runtime.powerState)
+        try:
+            task = self.current_vm_obj.CreateScreenshot_Task()
+            wait_for_task(task)
+        except vim.fault.FileFault as e:
+            self.module.fail_json(msg="Failed to create screenshot due to errors when creating or accessing one or more"
+                                      " files needed for this operation, %s" % to_native(e.msg))
+        except vim.fault.InvalidState as e:
+            self.module.fail_json(msg="Failed to create screenshot due to VM is not ready to respond to such requests,"
+                                      " %s" % to_native(e.msg))
+        except vmodl.RuntimeFault as e:
+            self.module.fail_json(msg="Failed to create screenshot due to runtime fault, %s," % to_native(e.msg))
+        except vim.fault.TaskInProgress as e:
+            self.module.fail_json(msg="Failed to create screenshot due to VM is busy, %s" % to_native(e.msg))
+
+        if task.info.state == 'error':
+            return {'changed': self.change_detected, 'failed': True, 'msg': task.info.error.msg}
+        else:
+            download_file_size = None
+            self.change_detected = True
+            file_url = self.generate_http_access_url(task.info.result)
+            if self.params.get('local_path'):
+                if file_url:
+                    download_file_size = self.download_screenshot_file(file_url=file_url,
+                                                                       local_file_path=self.params['local_path'],
+                                                                       file_name=task.info.result.split('/')[-1])
+            screenshot_facts = self.get_screenshot_facts(task.info, file_url, download_file_size)
+            return {'changed': self.change_detected, 'failed': False, 'screenshot_facts': screenshot_facts}
```

Comment:
This is to align new Ansible facts guidelines - 

```suggestion
            return {'changed': self.change_detected, 'failed': False, 'screenshot_info': screenshot_facts}
```